NotAmyCodeReview.cpp

```diff
@@ -1 +1,83 @@
+struct Player
+{
+	char* name;
+	int id;
+};
 
+class PlayerManager
+{
+	Player** pPlayers;
+	int numPlayers;
+public:
+	PlayerManager() : numPlayers(0) {}
+	Player* CreatePlayer(const char* name, int id);
+	Player* GetPlayerById(int id);
+	int GetNumPlayers();
+	void DestroyPlayer(Player* pPlayer);
+	void DestroyPlayerById(int id);
+	void DestroyAllPlayers();
+};
+
+Player PlayerManager::CreatePlayer(const char* name, int id)
+{
+	// Create new Player
+	Player* newPlayer = new Player;
+	newPlayer.name = name;
+	newPlayer.id = id;
+
+	// Add to pPlayers array
+	Player** newpPlayers = new Player * [numPlayers+1]
```

Comment:
Copying over to a new array on every create isn't pretty, it also doesn't work well with deletes.
I would keep track of the max size of the array, and when you reach it, double the max (similar how other list implementations work)